ci-operator/step-registry/hypershift/openstack/create/hypershift-openstack-create-chain.yaml

```diff
@@ -0,0 +1,208 @@
+        --openstack-node-flavor ${OPENSTACK_COMPUTE_FLAVOR}
+        --openstack-node-image-name ${RHCOS_IMAGE_NAME}
+        --base-domain ${HYPERSHIFT_BASE_DOMAIN}
+        --control-plane-availability-policy ${HYPERSHIFT_CP_AVAILABILITY_POLICY}
+        --infra-availability-policy ${HYPERSHIFT_INFRA_AVAILABILITY_POLICY}
+        --pull-secret=/etc/ci-pull-credentials/.dockerconfigjson
+        --release-image ${RELEASE_IMAGE}
+        --annotations=hypershift.openshift.io/skip-release-image-validation=true
+      )
+      
+      if [[ $ENABLE_ICSP == "true" ]]; then
+        COMMAND+=(--image-content-sources "${SHARED_DIR}/mgmt_icsp.yaml")
+      fi
+      
+      if [[ -n $EXTRA_ARGS ]]; then
+        COMMAND+=(${EXTRA_ARGS})
+      fi
+     
+      if [[ $HYPERSHIFT_CREATE_CLUSTER_RENDER == "true" ]]; then
+        "${COMMAND[@]}" --render > "${SHARED_DIR}/hypershift_create_cluster_render.yaml"
+        exit 0
+      fi
+      
+      "${COMMAND[@]}"
+
+      set +e
+      export CLUSTER_NAME
+      oc wait --timeout=30m --for=condition=Available --namespace=clusters hostedcluster/${CLUSTER_NAME}
+      echo "Cluster became available, creating kubeconfig"
+      bin/hypershift create kubeconfig --namespace=clusters --name=${CLUSTER_NAME} >${SHARED_DIR}/nested_kubeconfig
+
+      export KUBECONFIG=${SHARED_DIR}/nested_kubeconfig
+      timeout 25m bash -c '
+        until [[ "$(oc -n openshift-ingress get service router-default -o jsonpath='{.status.loadBalancer.ingress[0].ip}')" != "" ]]; do
+            sleep 15
+        done
+      '
+      INGRESS_IP=$(oc -n openshift-ingress get service router-default -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+      echo ${INGRESS_IP} > ${SHARED_DIR}/INGRESS_IP
+      echo "Ingress IP was found: ${INGRESS_IP}"
+    from: hypershift-operator
+    grace_period: 5m0s
+    resources:
+      requests:
+        cpu: 100m
+        memory: 100Mi
+    timeout: 45m0s
+    credentials:
+    - mount_path: /etc/ci-pull-credentials
+      name: ci-pull-credentials
+      namespace: test-credentials
+    dependencies:
+    - name: "release:latest"
+      env: RELEASE_IMAGE_LATEST
+    - name: hypershift-operator
+      env: HYPERSHIFT_RELEASE_LATEST
+  - ref: openstack-provision-dnsrecords
+  - as: hostedcluster-healthy
+    cli: latest
+    commands: |-
+      set -exuo pipefail
+      
+      # For disconnected or otherwise unreachable environments, we want to
+      # have steps use an HTTP(S) proxy to reach the API server. This proxy
+      # configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+      # environment variables, as well as their lowercase equivalents (note
+      # that libcurl doesn't recognize the uppercase variables).
+      if test -f "${SHARED_DIR}/proxy-conf.sh"
+      then
+        # shellcheck disable=SC1090
+        source "${SHARED_DIR}/proxy-conf.sh"
+      fi
+
+      CLUSTER_NAME="$(echo -n $PROW_JOB_ID|sha256sum|cut -c-20)"
+      set +e
```

Comment:
`+e` is a code smell. What is the command that may fail? Is there no way to explicitly handle the potential error?

Reply:
I'll keep this one though because the goal is to write the junit if the `until` loop failed. I think it's fine. 

Also stolen from other platforms:

https://github.com/openshift/release/blob/1771d8a973f464bdf44375c8ed392c5dff372b71/ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml#L107-L115